### PR TITLE
Update govuk-frontend to the most recent version (2.4.0)

### PR DESCRIPTION
**Pull request by willc-work**
Update govuk to the most recent version

[Link to story](https://dsdmoj.atlassian.net/browse/AP-219)

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unneccessary whitespace changes. These make diffs harder to read and conflicts more likely. 
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by reggieb** [comment]
The main issues should be as described in the [2.0.0 release notes](https://github.com/alphagov/govuk-frontend/releases/tag/v2.0.0). Looking through them the only obvious issue I could find is:

- **Remove the width declaration from the `<select>` component** - I think `api/app/views/providers/address_selections/show.html.erb:47` applies a width style to a `select` tag.

I think the main vulnerability will be when behaviour is modified via JavaScript, and that that will be most apparent in cross browser tests - particularly in how pages behave with changing window sizes.

The current page that seems to rely most heavily on behaviour changes due to window sizes is `check_provider_answers`. If you look at `app/assets/stylesheets/providers/check-your-answers.scss` it contains more calls to dynamic SASS method like `govuk-media-query` than the rest of the app put together.

The elements that are most likely to be problematic are error and hint elements in forms. As the most complicated form is probably the `/providers/applicant` form, I'd suggest testing the behaviour of that when an error is shown, and the window size is changed - across a number of browsers.

In summary - I don't think the update will cause any significant problems. However, I recommend:

- [ ] remove the width class declaration in the select element highlighted above.
- [ ] test the app in a number of browsers, paying particular attention to `providers/check_provider_answers` and `providers/applicants` (the latter with errors being displayed (submit the form with no input)
